src_blog/_posts/Library-Preview-Regex.md

```diff
@@ -0,0 +1,64 @@
+
+I am proud to announce the release of the [`Regex` module](/docs/stdlib/regex) in Grain's standard library as of version `0.4.3`. This milestone is particularly exciting for a number of reasons:
+
+First, most overtly, this module provides an easy way of matching simple patterns in strings:
+
+```grain
+import Regex from "regex"
+import Result from "result"
+import Option from "option"
+
+// Grain's Result type tells us whether we made a mistake
+// in our pattern. We use Result.expect to unwrap the
+// parsed pattern or throw an error if it failed to parse.
+let pat = Result.expect(
+  "Failed to parse regular expression",
+  Regex.make("ca+[rt]"),
+)
+
+// Now that we've compiled our Regular Expression pattern,
+// we can use it to search strings. Because a pattern may
+// or may not match the input string, it returns an
+// Option&lt;Regex.MatchResult&gt;. Here, we know it matches,
+// so we can use Option.unwrap to get the underlying value.
+let matchResult = Option.unwrap(Regex.find(pat, "caar"))
+assert Option.unwrap(matchResult.group(0)) == "caar"
+
+// We can look at another example where we don't
+// match the whole string:
+let matchResult = Option.unwrap(Regex.find(pat, "scare"))
+assert Option.unwrap(matchResult.group(0)) == "car"
+
+// We can replace substrings too!
+assert Regex.replace(pat, "cart", "ski") == "skit"
+```
+
+The full documentation can be found [here](/docs/stdlib/regex), but this library should be very useful for building even more nifty features in the future, such as JSON support.
+
+In addition, this library marks a soft milestone in Grain's maturity: it's one of the most complex things which has been written in the language so far, and it is written entirely in standard Grain! It packs a regular expression parser (so we can understand what pattern you're trying to match) and matching engine, all written in a single Grain module with no fancy flags, foreign functions, or attributes! It is far and away the largest module in the standard library:
+
+```bash
+ $ find stdlib -name '*.gr' \
+   | xargs -L 1 wc -l \
+   | sort -nk1 | tail -5
+    1324 stdlib/runtime/numberUtils.gr
+    1465 stdlib/runtime/numbers.gr
+    1612 stdlib/string.gr
+    1709 stdlib/sys/file.gr
+    3055 stdlib/regex.gr
+```
+
+This module landing in the standard library demonstrates that writing complex programs entirely in Grain is possible. Grain's standard library still has a long way to go, but we're on our way (and maybe [you can even help](https://github.com/grain-lang/grain))!
```

Comment:
Should we link to the issues with the stdlib tag?

Reply:
That could be a good idea. @ospencer, thoughts?

Reply:
That'd require us to actually have stdlib issues created :P 

I think it's a good idea. I can do a pass of things we want people to implement.

Reply:
Idea to end on more positive framing:
```suggestion
This module landing in the standard library continues to demonstrate our vision—writing complex programs in Grain is both possible and fun, but the standard library does the hard work so you don't have to. We're constantly improving Grain's standard library and [you can help too](https://github.com/grain-lang/grain)!
```